### PR TITLE
AUTH-9 # Fixed issues with Node 4.x support

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: lib/login-providers/browser.js [hunk 1]
```diff
@@ -76,7 +76,8 @@ class BrowserLoginProvider extends LoginProviderBase {
             reject(body.error_description);
             return;
           }
-          resolve(super.storeJWT(body.id_token));
+          // Must use this.storeJWT instead of super.storeJWT to support bug in Node 4.
+          resolve(this.storeJWT(body.id_token));
```
Comment (jokeyrhyme) [inline lib/login-providers/browser.js:80]: This is interesting. I thought Node 4.x supported `super` in methods. At least it does according to http://node.green/

Did you change this for some other reason or is this definitely breaking tests?

File: lib/common/tenant.js [hunk 2]
```diff
@@ -64,7 +64,8 @@ function remove (tenantName) {
 }
 
 module.exports = {
-  get,
-  set,
+  // Can not use same syntax as remove below to support Node 4.
+  get: get,
+  set: set,
```
Comment (jokeyrhyme) [inline lib/common/tenant.js:69]: This is interesting. I thought Node 4.x supported object shorthand properties. At least, it does according to http://node.green/

It's also super weird that `{ get: get }` is broken, but `{ remove }` is not. Perhaps this is more to do with shorthand getter|setter methods than it is to do with shorthand properties? /shrug

**Comment by jokeyrhyme** [comment]
Assuming this now passes tests: LGTM 👍

File: lib/login-providers/browser.js (hunk 1 above)
Comment (mymattcarroll) [reply lib/login-providers/browser.js:80]: This was breaking tests: http://stackoverflow.com/questions/32943776/using-super-within-an-arrow-function-within-an-arrow-function-within-a-method

File: lib/common/tenant.js (hunk 2 above)
Comment (mymattcarroll) [reply lib/common/tenant.js:69]: Yeah, I assume it has something to do with the `get` and `set` keywords

File: lib/login-providers/browser.js (hunk 1 above)
Comment (jokeyrhyme) [reply lib/login-providers/browser.js:80]: I know that arrow functions don't specify their own `this` or `super`, so these keywords refer to the `this` or `super` from the next outer closure (if any). I had no idea this was broken in V8 for a while there. Fascinating. :D

**Pull request merged**